src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -740,3 +796,58 @@ interface CursorPos {
   x: number;
   y: number;
 }
+
+class Rectangle {
+  imageWidth: number;
+  imageHeight: number;
+  brushRadius: number;
+  // Lowest y touched by the brush.
+  top: number;
+  // Highest y touched by the brush. 
+  bottom: number;
+  // Lowest x touched by the brush.
+  left: number;
+  // Highest x touched by the brush.
+  right: number;
+  
+  constructor(imageWidth: number, imageHeight: number, brushWidth: number, coord: Coordinate) {
+    this.imageWidth = imageWidth;
+    this.imageHeight = imageHeight;
+    this.brushRadius = brushWidth / 2;
+
+    this.top = (coord.y - this.brushRadius) >= 0 ? (coord.y - this.brushRadius) : 0;
```

Comment:
I always find these if statements hard to read and find `max(coord.y - this.brushRadius, 0);` easier to read.